packages/dds/merge-tree/src/mergeTreeDeltaCallback.ts

```diff
@@ -15,15 +15,31 @@ import { ISegment } from "./mergeTree";
 export type MergeTreeDeltaOperationType =
     typeof MergeTreeDeltaType.ANNOTATE | typeof MergeTreeDeltaType.INSERT | typeof MergeTreeDeltaType.REMOVE;
 
-// Note: Assigned negative integers to avoid clashing with MergeTreeDeltaType
+/**
+ * Enum-like constant defining the types of "maintenance" events on a merge tree.
+ * Maintenance events correspond to structural segment changes or acks of pending segments.
+ *
+ * Note: these values are assigned negative integers to avoid clashing with `MergeTreeDeltaType`.
+ */
 export const MergeTreeMaintenanceType = {
+    /**
+     * Notification that a segment "append" has occurred, i.e. two adjacent segments have been merged.
+     * BEWARE: `deltaSegments` on the corresponding event will contain both the merged segment and the latter
```

Comment:
felt worth calling this out b/c it's the only event here which displays "stale" segments as far as I know. Seems like something we might want to fix. It is used by one of our examples it seems.